data-explorer/kusto/query/toscalarfunction.md

```diff
@@ -31,10 +31,43 @@ If the result is a tabular, then the first column and first row will be taken fo
 > [!TIP]
 > You can use a [let statement](letstatement.md) for readability of the query when using `toscalar()`.
 
-**Notes**
+## Limitations
+
+`toscalar()` can be calculated a constant number of times during the query execution which means that the function can't be applied on row-level (for-each-row scenario).
+Usually, when this limitation is hit, the following error will be reported: `can't use '<column name>' as it is defined outside its row-context scope.`
```

Comment:
```suggestion
`toscalar()` can't be applied on a scenario that applies the function on each row. This is because the function can only be calculated a constant number of times during the query execution.
Usually, when this limitation is hit, the following error will be returned: `can't use '<column name>' as it is defined outside its row-context scope.`
```